kucher/view/main_window/register_view_widget/import_export_dialog.py

```diff
@@ -0,0 +1,302 @@
+#
+# This file is part of Kucher.
+# Kucher is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+# Kucher is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty
+# of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+# You should have received a copy of the GNU General Public License along with Kucher.
+# If not, see <http://www.gnu.org/licenses/>.
+#
+# Author: Pavel Kirienko <pavel.kirienko@zubax.com>
+#
+
+import os
+import yaml
+import asyncio
+import math
+import itertools
+import enum
+import typing
+
+from logging import getLogger
+from PyQt5.QtWidgets import QWidget, QFileDialog, QTableWidget, QTableWidgetItem, QDialog, QLabel, \
+    QPushButton, QHeaderView, QMessageBox
+from popcop.standard.register import ValueType
+
+from kucher.view.device_model_representation import Register
+from kucher.view.utils import show_error, lay_out_horizontally, lay_out_vertically, get_monospace_font
+
+_logger = getLogger(__name__)
+
+
+class CheckResult(enum.Enum):
+    NOT_MUTABLE = enum.auto()
+    INCORRECT_TYPE = enum.auto()
+    INCORRECT_DIMENSION = enum.auto()
+    OUTSIDE_RANGE = enum.auto()
+    UNKNOWN = enum.auto()
+    NO_ERROR = enum.auto()
+
+
+CHECK_RESULT_MAPPING: typing.Dict[CheckResult, str] = {
+    CheckResult.NOT_MUTABLE:         'This parameter cannot be modified on this device: ',
+    CheckResult.INCORRECT_TYPE:      'This parameter type is incorrect: ',
+    CheckResult.INCORRECT_DIMENSION: 'This parameter dimension is incorrect: ',
+    CheckResult.OUTSIDE_RANGE:       'This parameter value is outside permitted range: ',
+    CheckResult.UNKNOWN:             'Reason: ',
+}
+
+
+def export_registers(parent, registers):
+    """
+    Store all mutable and persistent registers in a .yml file
+    """
+    dialog_box = QWidget()
+    dialog_box.setGeometry(10, 10, 1000, 700)
+
+    file_name, _ = QFileDialog.getSaveFileName(dialog_box, 'Export configuration file', 'config.yml',
+                                               'YAML Files (*.yml  *.yaml);;All Files (*)')
+    if not file_name:
+        return
+    try:
+        _file = open(file_name, 'w+')
+    except Exception as ex:
+        show_error('Export failed',
+                   f'Cannot open {file_name}',
+                   f'Error: {str(ex)}',
+                   parent)
+        _logger.exception(f'File {file_name} could not be open: {str(ex)}')
+        return
+
+    async def executor():
+        try:
+            register_yaml = {}
+            for reg in registers:
+                if reg.mutable and reg.persistent:
+                    try:
+                        register_yaml[reg.name] = await reg.read_through()
+                    except Exception as ex:
```

Comment:
In case of an error, you just stop the loop and return. Makes no sense handling the exception inside the loop as it requires you to manage the control flow manually (see the explicit `return` below). You can simplify this by moving the except clause outside of the loop, just above `finally` on the same level.